fetch.bs

```diff
@@ -3702,132 +3776,145 @@ Range Requests</cite>. [[HTTP-RANGE]] However, this is not widely supported by b
 
-      <p class="note no-backref">This makes the `<code>Content-Length</code>`
-      <a for=/>header</a> unreliable to the extent that it was reliable
-      to begin with.
+       <li><p>Let <var>codings</var> be the result of <a>extracting header list values</a> given
+       `<code>Content-Encoding</code>` and <var>response</var>'s <a for=response>header list</a>.
 
-     <li>
-      <p><a for=ReadableStream>Enqueue</a> a <code>Uint8Array</code>
-      object wrapping an <code>ArrayBuffer</code> containing <var>bytes</var> to <var>stream</var>.
-      If that threw an exception, <a lt=terminated for=fetch>terminate</a> the ongoing
-      fetch with <i>fatal</i>, <a abstract-op>error</a>
-      <var>stream</var> with that exception and abort these subsubsteps.
-
-     <li><p>If <var>stream</var> doesn't <a lt="need more data" for=ReadableStream>need more
-     data</a> and <var>request</var>'s <a>synchronous flag</a> is unset, ask the user agent
-     to <a for=fetch>suspend</a> the ongoing fetch.
-    </ol>
+       <li>
+        <p>Set <var>bytes</var> to the result of <a lt="handle content codings">handling content
+        codings</a> given <var>codings</var> and <var>bytes</var>.
 
-   <li><p>If at any point the bytes transmission for <var>response</var>'s message body is done
-   normally and <var>stream</var> is <a for=ReadableStream>readable</a>,
-   then <a abstract-op>close</a> <var>stream</var>.
-   <!-- XXX xref message body -->
+        <p class="note no-backref">This makes the `<code>Content-Length</code>` <a for=/>header</a>
+        unreliable to the extent that it was reliable to begin with.
 
-   <li>
-    <p>If at any point <a for=/>fetch</a> is
-    <a for=fetch>terminated</a> with reason <var>reason</var>,
-    run these subsubsteps:
+       <li>
+        <p><a for=ReadableStream>Enqueue</a> a <code>Uint8Array</code> object wrapping an
+        <code>ArrayBuffer</code> containing <var>bytes</var> to <var>stream</var>. If that threw an
+        exception, <a lt=terminated for=fetch>terminate</a> the ongoing fetch with <i>fatal</i>, <a
+        abstract-op>error</a> <var>stream</var> with that exception.
 
-    <ol>
-     <li><p>Set <var>response</var>'s <a lt="termination reason" for=response>termination
-     reason</a> to <var>reason</var>.
+       <li><p>If <var>stream</var> doesn't <a lt="need more data" for=ReadableStream>need more
+       data</a> and <var>request</var>'s <a>synchronous flag</a> is unset, ask the user agent to <a
+       for=fetch>suspend</a> the ongoing fetch.
+      </ol>
 
-     <li><p>If <var>stream</var> is <a for=ReadableStream>readable</a>,
-     <a abstract-op>error</a> <var>stream</var> with a
-     <code>TypeError</code>.
+     <li><p>Otherwise, if the bytes transmission for <var>response</var>'s message body is done
+     normally and <var>stream</var> is <a for=ReadableStream>readable</a>, then
+     <a abstract-op>close</a> <var>stream</var> and abort these steps.
```

Comment:
abort these steps -> break

Reply:
Right, this is the root of the confusion. I don't want to break, I want to abort the whole "in parallel" block.

I thought "abort these steps" was scoped to the "in parallel" block, since it says "run these steps", whereas "while true" doesn't reference steps, so I didn't think it was a separate set of steps.

Reply:
Oh, well, "abort these steps" is generally pretty unclear, which is why we're trying to move away from it. So maybe "Abort the in parallel steps", or set a boolean and use that to guard subsequent steps.

Also, it seems you're forgetting to close the connection, if you truly do abort the steps.